### DISCUSSION TITLE
how to remove nested josn element using request-transformer

### DISCUSSION THREAD

**Discussion by RahulRdec** [Kong/kong#7863] [Help] [answered]
I am able to remove the top level element("kind") but can't remove nested. I want to remove namespace from the below request. I am trying 'spec.namespace' in remove body but isn’t working. any help will be appreciated. Thanks.

"kind":"update"
"spec": {"namespace": "teamname",}

**Comment by fffonion** [comment]
@RahulRdec Could you post a sample of JSON you are transforming and the request-transformer plugin's full config?

**Comment by RahulRdec** [comment]
@fffonion  In plugin I am only populating remove-->body param. I have tried multiple options - spec.namepsce & spec.namepace: etc but nothing seems working.

Request:
{
  "kind": "namespace",
  "action": "create",
  "spec": {
    "namespace": "teamname",
    "sloEmail": "bossman@xyz.com",
    "teamEmailDL": "TeamDistroList@xyx.com",
    "askId": "ABCWM",
    "adGroup": "myexample_adgroup"}
}

**Comment by fffonion** [comment, marked as answer]
@RahulRdec I see, i missed the `nested` fields parts. Those are not supported by request transformer plugin. You can write your
own plugin that does the transform, or use the coming `jq` plugin (which is included in the enterprise offering and aims on more `jq`-style JSON transformation).

**Reply by RahulRdec** [reply]
Thanks @fffonion . That helps.
